src/audio/kpb.c

```diff
@@ -998,18 +998,25 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 
 		} while (buff != first_buff);
 
-		/* Calculate time in clock ticks each draining event shall
-		 * take place. This time will be used to synchronize us with
-		 * an end application interrupts.
+		/* Should we drain in synchronized mode (sync_draining_mode)?
+		 * Note! We have already verified host params during
+		 * kpb_prepare().
 		 */
-
-		drain_interval = (host_period_size / bytes_per_ms) *
-				 ticks_per_ms;
-		/* In draining intervals we will fill only two periods
-		 * and give host time to read it.
-		 * This way we are safe to not overflow host buffer.
-		 */
-		period_bytes_limit = host_period_size * 2;
+		if (2 * kpb->hb_buffer_size > host_buffer_size) {
```

Comment:
Where does 2 come from ? We've just removed the comment. We should really make this a macro too.

Reply:
Because we want want host buffer to be much bigger (two times in this case, 4,2 seconds in total) than firmware buffer to drain with unlimited speed and no risk buffer overwrite in case of slow host wake up from some power state change.

Reply:
@mrajwa Comment should be there in code explaining the 2, as it's not obvious to those without expertise in this.

Reply:
Yep, I will update it to be more understandable.